e2e/nomostest/nt.go

```diff
@@ -810,11 +810,14 @@ func (nt *NT) setupConfigConnector() {
 			},
 		},
 	}
-	nt.T.Cleanup(func() {
-		if err := nt.KubeClient.Delete(kccObj); err != nil {
-			nt.T.Error(err)
-		}
-	})
+	// TODO(sdowell): there are currently issues with cleanup, most likely related to a
```

Comment:
The symptom is that there are leftover KCC managed resources on the cluster which are stuck in a finalizing state (e.g. IAMPolicyMember). 

The behavior previously was to install the ConfigConnector resource as a one off with the provisioning script, whereas my recent change made it installed/uninstalled with every test execution. It seems there's most likely an issue with waiting for finalizers to complete, e.g.
- the CS finalizer deletes the object but doesn't wait for not found
- the test framework deletes the ConfigConnector object before the `cnrm.cloud.google.com/finalizer` finalizer completes
- the resource gets stuck in a finalizing state

Reply:
so you're planning to uncomment this when the finalizer is fixed?

Reply:
Ideally, yes. But I'll need to make sure the cleanup logic is working properly. We might need some additional wait logic added

Reply:
Yes. It's possible this need to be move into the Reset and/or Clean code so it can be handled in order and not happen if a pre-requisite delete errors.

Reply:
Actually... is there a status on this object that says whether any KCC objects exist in the namespace? Maybe we should ask the KCC team for that. It would make it easier to determine when it's safe to delete.

Reply:
This is invoked after reset/cleanup, as `setupConfigConnector` is called first and `Cleanup` behaves like a stack. But agreed that we should wait for a signal that the resource is safe to delete, then delete it, then wait for a signal that deletion has reconciled.